public/js/fxa-analytics.js

```diff
@@ -49,15 +49,38 @@ async function sendPing(el, eventAction, eventLabel = null) {
   }
 }
 
+
 function getFxaUtms(url) {
-  const utmSource = encodeURIComponent(document.body.dataset.serverUrl.replace(/(^\w+:|^)\/\//g, ""));
-  url.searchParams.append("utm_source", utmSource);
-  url.searchParams.append("utm_campaign", document.body.dataset.utmCampaign);
+  const bodyDataset = document.body.dataset;
+  getUTMNames().forEach(param => {
+    if (bodyDataset[param]) {
+      url.searchParams.append(param, encodeURIComponent(bodyDataset[param]));
+    }
+  });
   url.searchParams.append("form_type", "email");
   return url;
 }
 
+function saveReferringPageData(utmParams) {
+  const bodyDataset = document.body.dataset;
+  getUTMNames().forEach(param => {
+    if (utmParams.has(param)) {
+      const cleanParam = utmParams.get(param);
+      bodyDataset[param] = cleanParam.replace(/[&<>"',.`=:/]/g, "");
+    }
+  });
+}
+
+function getUTMNames() {
+  return ["utm_source", "utm_medium", "utm_campaign", "utm_term", "utm_content"];
```

Comment:
Is this everything? Or should we do some generic parser that looks for any query param that starts with `utm_*`?
Sorry, I'm not a GA expert and don't know how flexible these query params are.

Reply:
This is a good point. These were the parameters we were specifically asked to forward in #844 and my thinking was that limiting the parameters to just these would prevent us from potentially keeping and forwarding a bunch of unnecessary junk.